src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -804,8 +804,43 @@ protected bool TakeStartLine(SocketInput input)
                 QueryString = queryString;
                 HttpVersion = httpVersion;
 
-                bool caseMatches;
+                if (requestUrlPath.Length > 0 && requestUrlPath[0] != '/')
+                {
+                    int hostIndex;
+                    if (requestUrlPath.StartsWith("http://"))
+                    {
+                        hostIndex = 7;
+                    }
+                    else if (requestUrlPath.StartsWith("https://"))
+                    {
+                        hostIndex = 8;
+                    }
+                    else
+                    {
+                        ReportCorruptedHttpRequest(new BadHttpRequestException($"Invalid request path: {requestUrlPath}"));
+                        return true;
+                    }
+
+                    int pathIndex = requestUrlPath.IndexOf('/', hostIndex);
+                    if (pathIndex == -1)
+                    {
+                        int queryIndex = requestUrlPath.IndexOf('?', hostIndex);
```

Comment:
Hmm, looking at the code that brings up an interesting point. pathBegin and pathEnd are now the wrong name, it should be urlBegin or targetBegin. Similar for requestUrlPath.